nautobot/docs/development/core/local-k8s.md

```diff
@@ -0,0 +1,202 @@
+# Using Local Kubernetes Cluster
+
+This guide will help you set up a local Kubernetes cluster using [Kind](https://kind.sigs.k8s.io/) and connect running Nautobot containers to the Kind network.
+
+## Kind Installation
+
+In this chapter we will install Kind on your local machine and create a new Kind Kubernetes cluster.
+
+To install Kind on macOS:
+
+```bash
+brew install kind
+```
+
+To install Kind on Linux:
+
+```bash
+curl -fsSL \
+    "https://kind.sigs.k8s.io/dl/latest/kind-linux-amd64" \
+    --output /usr/local/bin/kind
+chmod +x /usr/local/bin/kind
+```
+
+To create a new Kind cluster:
+
+```bash
+kind create cluster \
+    --name=nautobot \
+    --config=development/kind-config.yaml
+```
+
+Expected output shows the cluster is created:
+
+```text
+Creating cluster "nautobot" ...
+ ✓ Ensuring node image (kindest/node:v1.31.0) 🖼
+ ✓ Preparing nodes 📦
+ ✓ Writing configuration 📜
+ ✓ Starting control-plane 🕹️
+ ✓ Installing CNI 🔌
+ ✓ Installing StorageClass 💾
+Set kubectl context to "kind-nautobot"
+You can now use your cluster with:
+
+kubectl cluster-info --context kind-nautobot
+
+Have a question, bug, or feature request? Let us know! https://kind.sigs.k8s.io/#community 🙂
+```
+
+To export the `kubeconfig` file with an internal cluster IP address for access from connected Nautobot containers, run the following command:
+
+```bash
+kind get kubeconfig \
+    --internal \
+    --name=nautobot \
+    > development/kind-kube-config
+```
+
+## Connecting Nautobot Containers
+
+This chapter will help you connect running Nautobot containers to the Kind network.
+
+!!! Note
+    These steps needs to be done every time you start the Nautobot containers.
```

Comment:
Does `kind get kubeconfig` above also need to be done every time, or is it a one-off?

Reply:
When you generate kubeconfig, you can store it to a location, and then use `export KUBECONFIG=path/to/kubeconfig/file` to set the "current" kubeconfig you want to use.

Reply:
Is there a security implication to committing the kubeconfig to git, or is there some other reason the user needs to manually (re)generate this file locally?

Reply:
For each `kind create cluster`, there is a different server certificate generated and thus a different kube config file.

Would rather prepare something like `invoke create-dev-k8s-cluster` task to automate these steps.

Reply:
Ok, so if I recreate the cluster, I need to regenerate the file? That's an important thing to document. :-)

Reply:
Separated those two steps into a chapter.